src/handlers/input.rs

```diff
@@ -121,6 +94,74 @@ pub fn handler(key: Key, app: &mut App) {
   }
 }
 
+fn process_input(app: &mut App, input: String) {
+  // Don't do anything if there is no input
+  if input.is_empty() {
+    return;
+  }
+
+  // On searching for a track, clear the playlist selection
+  app.selected_playlist_index = Some(0);
+
+  if attempt_process_uri(app, &input, "https://open.spotify.com/", "/")
+    || attempt_process_uri(app, &input, "spotify:", ":")
+  {
+    return;
+  }
+
+  // Default fallback behavior: treat the input as a raw search phrase.
+  app.dispatch(IoEvent::GetSearchResults(input, app.get_user_country()));
+  app.push_navigation_stack(RouteId::Search, ActiveBlock::SearchResultBlock);
+}
+
+fn spotify_resource_id(base: &str, uri: &str, sep: &str, resource_type: &str) -> (String, bool) {
+  let uri_prefix = format!("{}{}{}", base, resource_type, sep);
+  let id_string_with_query_params = uri.trim_start_matches(&uri_prefix);
+  let query_idx = id_string_with_query_params
```

Comment:
I've made a source-code comment on this, but at some point we may be re-inventing the wheel with regards to parsing URIs.

At the moment, my strong belief is that this should actually be pretty robust and handle all cases of Spotify share URLs, but in the case that it doesn't we should introduce the `url` crate to parse out share-URLs as a special case.

Reply:
We can add the `url` crate if it makes things easier, but this this looks fine to me.